protostar/migrator/maybe_tmp_directory.py

```diff
@@ -1,29 +1,21 @@
-from contextlib import AbstractContextManager
-from os import listdir, rmdir
+from contextlib import AbstractContextManager, contextmanager
 from pathlib import Path
 from typing import Union
 
 from protostar.protostar_exception import ProtostarException
 
 
-class MaybeTmpDirectory(AbstractContextManager):
-    def __init__(self, path: Path):
-        self._path = path
+@contextmanager
+def MaybeTmpDirectory(path: Path):
+    if path.exists():
+        raise DirectoryExistsException(path)
 
-    def __enter__(self) -> None:
-        self._create_dir()
+    path.mkdir()
+    yield path
```

Comment:
I think you have to put this in `try...finally` in order to remove the directory in case of exceptions